File: src/utils/platform.ts
import type { Platform } from '@/types/common'

/** 获取当前平台 */
export function getPlatform(): Platform {
  if (navigator.userAgent.includes('Win')) return 'windows'
  if (navigator.userAgent.includes('Mac')) return 'macos'
  return 'linux'
}

/** 是否为 Windows */
export function isWindows(): boolean {
  return getPlatform() === 'windows'
}

/** 是否为 macOS */
export function isMacOS(): boolean {
  return getPlatform() === 'macos'
}

File: src-tauri/src/utils/platform.rs
/// 平台相关工具函数

/// 是否为 Windows
pub fn is_windows() -> bool {
    cfg!(windows)
}

/// 是否为 macOS
pub fn is_macos() -> bool {
    cfg!(target_os = "macos")
}

/// 是否为 Linux
pub fn is_linux() -> bool {
    cfg!(target_os = "linux")
}

/// 获取平台名称
pub fn platform_name() -> &'static str {
    if is_windows() {
        "windows"
    } else if is_macos() {
        "macos"
    } else {
        "linux"
    }
}
